app/assets/javascripts/covid19/src/webapp/components/SVGProteoma.tsx

```diff
@@ -0,0 +1,496 @@
+                                classStyle="pink"
+                                def="M303.73,437.25c-2,6.16-3.67,12.51-5.05,18.9l-46.88-10c1.61-7.36,4.09-16.64,6.35-23.75Z"
+                            />
+                            <ProteomaPath
+                                stateSetters={stateSetters}
+                                name="M"
+                                classStyle="red"
+                                def="M298.27,458.11c-1.08,5.18-1.95,10.47-2.62,15.76L248.06,468c.88-7.06,1.84-12.8,3.32-19.87Z"
+                            />
+                            <ProteomaPath
+                                stateSetters={stateSetters}
+                                name="E"
+                                classStyle="orange"
+                                def="M295.41,475.85c-.62,5.29-1,10.64-1.24,15.93l-47.94-1.67c.27-7.16.72-13,1.58-20.11Z"
+                            />
+                            <ProteomaPath
+                                stateSetters={stateSetters}
+                                name="ORF3"
+                                classStyle="gray"
+                                def="M295.84,527.65l-47.47,6.67a232.73,232.73,0,0,1-2.21-42.22l47.94,1.67c-.07,2-.1,4.09-.1,6.23A208.57,208.57,0,0,0,295.84,527.65Z"
+                            />
+                            <ProteomaPath
+                                stateSetters={stateSetters}
+                                name="Spike"
+                                classStyle="green"
+                                def="M364.09,654.8,332.63,691c-24.05-21.05-42.54-43.79-56.48-69.48s-22.95-53.57-27.5-85.2l47.47-6.67A205.66,205.66,0,0,0,364.09,654.8Z"
+                            />
+                        </g>
+                    </g>
+                </SVG>
+            </Layer>
+            <span>
+                {loading &&
+                    i18n.t("Searching for {{search}} in databases...", {
+                        nsSeparator: false,
+                        search: search,
+                    })}
+            </span>
+        </Container>
+    );
+});
+
+interface ProteomaPathProps {
+    name: string;
+    classStyle: string;
+    def: string;
+    stateSetters: {
+        setSearch: (value: string) => void;
+        setTitle: (value: React.ReactNode) => void;
+        setProteomaSelected: (value: boolean) => void;
+        setLoading: (value: boolean) => void;
+    };
+}
+
+const ProteomaPath: React.FC<ProteomaPathProps> = React.memo(props => {
+    const {
+        name,
+        classStyle,
+        def,
+        stateSetters: { setTitle, setSearch, setProteomaSelected, setLoading },
+    } = props;
+
+    return (
+        <path
+            className={classStyle}
+            d={def}
+            onMouseEnter={() => setTitle(<span>{name}</span>)}
+            onClick={() => {
+                setLoading(true);
+                setSearch(name);
+                setLoading(false);
```

Comment:
@tokland How can I retrieve when the DataGrid has stopped looking for something?